### PR TITLE
Add perimeter_ext function to LineMeasurableExt trait

### PR DESCRIPTION
**Pull request by zhangfengcdt**
Implements a Perimeter trait for the geo-generic-alg crate that calculates the perimeter of geometries using generic WKB traits, following the same pattern as the existing Area trait.

- Follows the same generic WKB transformation pattern as area.rs
- Uses CoordFloat bounds for floating-point calculations with hypot() method
- Comprehensive test coverage matching area.rs patterns

This provides a cleaner API where users can call .perimeter() directly on polygons instead of manually summing exterior and interior ring lengths.

Test Coverage
- All geometry types with various configurations
- Edge cases (empty geometries, single points, degenerate cases)
- Numerical stability testing with large coordinate shifts
- Complex polygons with manual calculation verification
- WKT parsing integration
- Mixed geometry collections

### CODE REVIEW

File: geo-generic-alg/src/algorithm/perimeter.rs [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+fn linestring_perimeter<T, LS: LineStringTraitExt<T = T>>(linestring: &LS) -> T
+where
+    T: CoordFloat,
+{
+    let mut perimeter = T::zero();
+    for line in linestring.lines() {
+        let start_coord = line.start_coord();
+        let end_coord = line.end_coord();
+        let delta = end_coord - start_coord;
+        perimeter = perimeter + delta.x.hypot(delta.y);
```
Comment (Kontinuation) [inline geo-generic-alg/src/algorithm/perimeter.rs]: We have implemented `perimeter_trait` for `LineTraitExt`. Why not directly call `line.perimeter_trait()`?

File: geo-generic-alg/src/algorithm/perimeter.rs [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+impl<T, L: LineTraitExt<T = T>> PerimeterTrait<T, LineTag> for L
+where
+    T: CoordFloat,
+{
+    fn perimeter_trait(&self) -> T {
+        // For a line, return its length as the perimeter
+        let start_coord = self.start_coord();
+        let end_coord = self.end_coord();
+        let delta = end_coord - start_coord;
+        delta.x.hypot(delta.y)
```
Comment (Kontinuation) [inline geo-generic-alg/src/algorithm/perimeter.rs]: We can implement perimeter based on `LengthMeasurableExt`, which is a cleaner and more flexible approach, and also better fit with the overall design of geo line measurement algorithms.

File: geo-generic-alg/src/algorithm/perimeter.rs [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+                GeometryTypeExt::MultiPolygon(mp) => mp.perimeter_trait(),
+                GeometryTypeExt::GeometryCollection(gc) => gc.perimeter_trait(),
+                GeometryTypeExt::Rect(rect) => rect.perimeter_trait(),
+                GeometryTypeExt::Triangle(triangle) => triangle.perimeter_trait(),
+            })
+            .fold(T::zero(), |acc, next| acc + next)
+    }
+}
+
+// Critical: GeometryTag implementation for WKB compatibility
```
Comment (Kontinuation) [inline geo-generic-alg/src/algorithm/perimeter.rs]: This comment looks weird. There's nothing specific about WKB in this trait.

**Review by Kontinuation — APPROVED** [review]
I can bear with the chaos of our new line measurement related algorithms not being aligned with the original design of geo, but we may come back to rework this some day.

**Comment by zhangfengcdt** [comment]
> I can bear with the chaos of our new line measurement related algorithms not being aligned with the original design of geo, but we may come back to rework this some day.

Yes, thanks for the suggestion. It is actually much cleaner to add perimeter to LineMeasurableExt trait. I added a new perimeter_ext because I think the callers (e.g., st_length, st_perimeter) will still need to differentiate them. Also removed the Perimeter trait completely.
